test/unit/pdf/finance/invoice_report_data_test.rb

```diff
@@ -113,12 +113,15 @@ def setup
     default_options = Paperclip::Attachment.default_options
     Paperclip::Attachment.stubs(default_options: default_options.merge(storage: :s3))
     @provider.profile.update(logo: Rack::Test::UploadedFile.new(file_fixture('wide.jpg'), 'image/jpeg', true))
-    URI.expects(:open).with(regexp_matches(/\Ahttps.*\/profiles\/logos\/invoice\/wide.png\z/)).returns(File.open(file_fixture('wide.jpg')))
+
+    # the extension in the request is .png, because images are converted to png in the 'invoice' style
+    stub_request(:get, %r{\Ahttps.*/profiles/logos/invoice/wide.png\z})
+        .to_return(status: 200, body: File.open(file_fixture('wide.jpg')))
 
     logo_file = nil
     @data.with_logo do |logo|
       logo_file = logo
-      assert logo.is_a? File
+      assert logo.is_a? StringIO
```

Comment:
why does this change? In both cases we used `#open`

Reply:
This is because originally the `URI.open` was stubbed, and it returned an object that wasn't the object that `URI.open` would normally return.

Now, instead of stubbing the `URI.open`, the HTTP request is stubbed, so now the test simulates the real behavior more precisely.

Reply:
You can merge as far as I'm concerned.